### PR TITLE
Fix email search

### PR DESCRIPTION
**Pull request by Nil20**
## Description

Clearly describe what has been changed. Include relevant context or background.
Explain how the issue was fixed (if applicable) and the root cause.

Link this pull request to the GitHub issue (and optionally name the branch `ocrvs-<issue #>`)

## Checklist

- [ ] I have linked the correct Github issue under "Development"
- [ ] I have tested the changes locally, and written appropriate tests
- [ ] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [ ] I have updated the GitHub issue status accordingly

### CODE REVIEW

File: packages/user-mgnt/src/features/verifyUser/handler.ts [hunk 1]
```diff
@@ -45,19 +42,15 @@ export default async function verifyUserHandler(
...
     return badRequest()
   }
 
-  if (mobile) {
-    user = await User.findOne({ mobile })
-  } else {
-    user = await User.findOne({
-      emailForNotification: { $regex: `^${email}$`, $options: 'i' }
-    })
-  }
```
Comment (Nil20) [inline packages/user-mgnt/src/features/verifyUser/handler.ts:60]: The regex was probably used to achieve case-insensitive matching, which is unnecessary now since all stored emails are already normalized to lowercase at creation time.

File: packages/login/src/views/ResetCredentialsForm/AuthDetailsVerificationForm.tsx [hunk 2]
```diff
@@ -42,11 +42,22 @@ const AuthDetailsVerificationComponent = ({ intl }: WrappedComponentProps) => {
...
+    return (
+      location.state?.forgottenItem || sessionStorage.getItem('forgottenItem')
+    )
+  })
+
+  useEffect(() => {
+    if (location.state?.forgottenItem) {
+      sessionStorage.setItem('forgottenItem', location.state.forgottenItem)
+    }
+  }, [location.state])
```
Comment (Nil20) [inline packages/login/src/views/ResetCredentialsForm/AuthDetailsVerificationForm.tsx]: Previously, navigating back or refreshing cleared location.state, leaving forgottenItem as undefined.

Storing it in sessionStorage ensures the value survives browser navigation within the same tab/session without needing global state or query params.
